src/Features/CSharp/Portable/EditAndContinue/BreakpointSpans.cs

```diff
@@ -188,6 +188,27 @@ private static int GetEndPosition(SyntaxNodeOrToken nodeOrToken)
                         ? TryCreateSpanForSwitchLabel((SwitchLabelSyntax)node, position)
                         : CreateSpan(caseClause.WhenClause);
 
+                case SyntaxKind.SwitchExpressionArm:
+                    var switchArm = (SwitchExpressionArmSyntax)node;
+                    return CreateSpan((position <= switchArm.WhenClause?.FullSpan.End == true) ? switchArm.WhenClause : (SyntaxNode)switchArm.Expression);
+
+                case SyntaxKind.SwitchExpression when
+                    node is SwitchExpressionSyntax switchExpression &&
+                    switchExpression.Arms.Count > 0 &&
+                    position >= switchExpression.OpenBraceToken.Span.End &&
+                    position <= switchExpression.CloseBraceToken.Span.Start:
+                    // This can occur if the cursor is on a separator. Find the nearest switch arm.
+                    switchArm = switchExpression.Arms[0];
+                    for (int i = 1, n = switchExpression.Arms.Count; i < n; i++)
+                    {
+                        if (position >= switchExpression.Arms[i].FullSpan.Start)
+                            switchArm = switchExpression.Arms[i];
+                        else
+                            break;
+                    }
```

Comment:
note: we have BinarySearch helpers.  would make sense to use here, no?

Reply:
I don't see any such helpers that would be applicable (to a SeparaterSyntaxList) but I can use SeparatedSyntaxList.LastIndexOf.

---
In reply to: [398824655](https://github.com/dotnet/roslyn/pull/42711#discussion_r398824655) [](ancestors = 398824655)

Reply:
i think separated lists also expose just hte list of nodes.

But this is nbd.  i'm fine with the code as is